src/tiled/commandmanager.cpp

```diff
@@ -59,17 +59,37 @@ CommandDataModel *CommandManager::commandDataModel()
     return mModel;
 }
 
+void CommandManager::registerMenu(QMenu *menu)
+{
+    mMenus.append(menu);
```

Comment:
Making `populateMenu` private would require `updateActions` to be public. I would  then call `updateActions` each time the Command Dialog is opened.